flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaInvocationHandler.java

```diff
@@ -189,7 +191,49 @@ public void stop() {
 		rpcEndpoint.tell(Processing.STOP, ActorRef.noSender());
 	}
 
-	// ------------------------------------------------------------------------
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+
+		if (o == null) {
+			return false;
+		}
+
+		if(Proxy.isProxyClass(o.getClass())) {
+			return o.equals(this);
+		}
```

Comment:
@StephanEwen , as till said, the subject of this pr is misleading. I means When call connect method in AkkaRpcService using same address and same rpc gateway class, the returned gateways are equals instead of return same gateway. I changed the subject.